velox/docs/develop/cache.rst

```diff
@@ -0,0 +1,152 @@
+===========================
+AsyncDataCache (File Cache)
+===========================
+
+Background
+----------
+Velox provides transparent file cache (AsyncDataCache) to accelerate table scan through the hot data reuse and prefetch. The file cache is integrated with the memory system to achieve dynamic memory sharing between file cache and query memory. When a query fails to allocate memory, we retry the allocation by shrinking the file cache. Therefore, the file cache size is automatically adjusted in response to the query memory usage change. See `Memory Management - Velox Documentation <https://facebookincubator.github.io/velox/develop/memory.html>`_  for more information about Velox's file cache.
+
+Configuration Properties
+------------------------
+The AsyncDataCache can be enabled by setting the following config:
+
+.. code-block:: bash
+
+    async-data-cache-enabled=true
+
+
+Other Properties
+----------------
+There is a ``cache.no_retention`` session property in Velox that can be set to control if a query's cached data is retained or not after its execution.
+
+.. list-table::
+   :widths: 30 10 10 70
+   :header-rows: 1
+
+   * - Property Name
+     - Type
+     - Default Value
+     - Description
+   * - cache.no_retention
+     - bool
+     - false
+     - If set to true, evicts out a query scanned data out of in-memory cache right after the access, and also skips staging to the SSD cache.​
+
+Set the ``hive.node_scheduler_affinity`` session property accordingly to turn ON/OFF cache.no_retention.​
+
+.. code-block:: bash
+
+    SET SESSION hive.node_selection_strategy='NO_PREFERENCE'; // To turn cache.no_retention ON.​
+    SET SESSION hive.node_selection_strategy='SOFT_AFFINITY'; // To turn cache.no_retention OFF.​
+    SET SESSION hive.node_selection_strategy='HARD_AFFINITY'; // To turn cache.no_retention OFF.​
+
+
+=========
+SSD Cache
+=========
+
+Background
+----------
+The AsyncDataCache is configured to use SSD when provided.
+The SSD serves as an extension for the async data cache (file cache).
```

Comment:
We used "AsyncDataCache" as a term before so we should be consistent in the usage when referring to the same concept.

Reply:
Changed to use "in-memory file cache (AsyncDataCache)"

Reply:
Ok. But now it is weird. In line 55 we say " ... in-memory file cache (AsyncDataCache) ...". And now ion this line we reverse it by using " ... the async data cache (file cache) ... ". At this point we can just use the term "AsyncDataCache".

```
The SSD serves as an extension for the AsyncDataCache.
```